Xamarin.PropertyEditing.Tests/PropertyViewModelTests.cs

```diff
@@ -1175,6 +1175,64 @@ public void AutocompleteCancels ()
 			Assert.That (tcs.Task.IsCanceled, Is.True);
 		}
 
+		[Test]
+		public void AutocompleteResults1NotUpdatedWhen2ndSearchCancelled ()
+		{
+			var target = new object ();
+			var property = GetPropertyMock ();
+
+			var editor1 = new Mock<IObjectEditor> ();
+			editor1.SetupGet (e => e.Target).Returns (target);
+			SetupPropertySetAndGet (editor1, property.Object);
+
+			var tcs = new TaskCompletionSource<IReadOnlyList<string>> ();
+
+			string[] results = new[] { "Foo", "Bar", "Baz" };
+			string[] results2 = new[] { "Foo2", "Bar2", "Baz2" };
+			var changeResultSet = false;
+			var complete1 = editor1.As<ICompleteValues> ();
+			complete1.Setup (c => c.GetCompletionsAsync (property.Object, It.IsAny<string> (), It.IsAny<CancellationToken> ()))
+				.Returns<IPropertyInfo, string, CancellationToken> ((a, b, c) => {
+					if (!changeResultSet) {
+						tcs.TrySetResult (results);
+					} else {
+						tcs.TrySetResult (results2);
+					}
+					return tcs.Task;
+				 });
+
+			var editor2 = new Mock<IObjectEditor> ();
+			editor2.SetupGet (e => e.Target).Returns (target);
+			SetupPropertySetAndGet (editor2, property.Object);
+
+			var complete2 = editor2.As<ICompleteValues> ();
+			complete2.Setup (c => c.GetCompletionsAsync (property.Object, It.IsAny<string> (), It.IsAny<CancellationToken> ()))
+				.Returns<IPropertyInfo, string, CancellationToken> ((a, b, c) => {
+					c.Register (() => {
+						tcs.TrySetCanceled ();
+					});
+					return tcs.Task;
+				});
+
+			var mockProvider = new MockEditorProvider (editor1.Object);
+			var resources = new MockResourceProvider ();
+			var platform = new TargetPlatform (mockProvider, resources) {
+				SupportsCustomExpressions = true,
+			};
+
+			var vm = GetViewModel (platform, property.Object, new[] { editor1.Object, editor2.Object });
+			Assert.That (vm.SupportsAutocomplete, Is.True);
+
+			vm.PreviewCustomExpression = "preview";
+
+			changeResultSet = true;
```

Comment:
We should add an `Assume` here to ensure that the results are the first set we expect them to be.